### ISSUE TITLE
When I hover over a card, I should see a dropshadow with the specified styling

### ISSUE THREAD

**Issue by ibagha** [concourse/resource-types-website#98]
Link to Figma: https://www.figma.com/file/pmLRwbPKrwDAGn6XLDoUiA/Concourse-Resources?node-id=508%3A0

**Comment by ibagha** [comment]
Builds off the card skeleton mentioned in issue #84 

**Styling**
box-shadow: 0px 0px 3px rgba(0, 0, 0, 0.2), 0px 2px 3px rgba(0, 0, 0, 0.2), -5px 5px 8px rgba(90, 85, 85, 0.14);

**Comment by jomsie** [comment]
i spent a bunch of time yesterday trying to figure out how to do a pretty simple hover in elm, but it appears not to be supported in the style of the rest of our styles. so there are three options:

1. use elm-css dependency and re-write most of our current styles (introduce a new dependency, will take some time)
2. have a basic external style sheet for the box shadows (super easy but not testable)
3. add some logic to track whether or not a card is hovered and change the styles within elm, probably by adding model / update / view loop functionality (probably the right choice, a bit more work upfront, but we'll extend that model eventually to add all the card info, so it'll have to be done eventually)

talked to @YoussB and i'm gonna try doin' #3 on my freedom friday.

**Comment by jomsie** [comment]
once this hover is done i'll make a card related PR

**Comment by pivotal-jamie-klassen** [comment]
@jomsie yo, I'd definitely use https://github.com/rtfeldman/elm-css or (probably even better) https://github.com/mdgriffith/elm-ui for your views, if it's not too late.

They both are not very amenable to end-to-end testing, though. This is because they both use not-very-opaque types that are difficult to write tests against - for example, when I use something like `myDiv = Html.Styled.div [margin (px 12)] []` (these functions are in elm-css, but elm-ui has similar problems), there's no useful function that lets me extract the set of styles back out of it to write assertions with (the `Html` type in `elm/html` has a similar problem, but somebody has done a bunch of work to write the functions in https://github.com/elm-explorations/test/tree/master/src/Test/Html that help you make similar assertions with some confidence.

If you go down this path -- and I strongly suggest you do, because the whole model/update/view thing for hovering is, despite decent structure, still pretty painful in the Concourse codebase -- I'd be happy to give some advice/pair on testing strategy a bit. If you do it right, you can have great testability and a much more mature way of doing styles.

**Comment by jomsie** [comment]
@pivotal-jamie-klassen yo, thank you for this :D

1. gonna use elm-ui
2. gonna try to avoid update functions for as long as we can
3. gonna have view types, and switch into html (via elm-ui) as late as possible

**Comment by jomsie** [comment]
just following up on the triple box shadow hover state, https://github.com/rtfeldman/elm-css/issues/265, looks like elm-ui doesn't support loads of shadows!

i've got a solution in mind, but it's not perfect... we'll see how it goes! this comment is more for reference than anything.

**Comment by jomsie** [comment]
Deleted old message because I referenced the wrong issue, BUT, here's something.

elm-ui only supports a single shadow (or more specifically, an inner and an outer shadow). Our hover has _3 layers_ of box-shadows.

Turns out compatibility mode (htmlAttribute) within a mouseover doesn't work.

I'm going to try to get the box-shadow as close as possible whist only using inner/outer shadows. I'll time box(shadow) it. 😆 😆 😆
